tests/drivers/uart/uart_basic_api/src/Makefile

```diff
@@ -1,3 +1,4 @@
 include $(ZEPHYR_BASE)/tests/Makefile.test
 
-obj-y += main.o test_uart_poll.o test_uart_fifo.o
+obj-y += main.o test_uart_poll.o
```

Comment:
the ztests are being run conditionally based on the CONFIG_UART_INTERRUPT_DRIVEN flag but we are compiling the test_uart_poll.o without any conditional check. this does not look logical to me.

Reply:
Had checked with the different uart drivers in code. Any basic uart driver supports the poll_in and poll_out callback implementation. Only the FIFO mode are encoded between the CONFIG_UART_INTERRUPT_DRIVEN macro. Also drivers that support FIFO mode also supports poll mode.

Reply:
Makefile and source code changes are not aligned. for the above make file changes the code should like below:

#ifdef CONFIG_UART_INTERRUPT_DRIVEN
  			 ztest_unit_test(test_uart_fifo_fill),
  			 ztest_unit_test(test_uart_poll_out));
 #endif
 			 ztest_unit_test(test_uart_poll_in),
 			 ztest_unit_test(test_uart_poll_out));

Reply:
@ramakrishnapallala  Code modified and posted for review